(window.webpackJsonp=window.webpackJsonp||[]).push([[28],{428:function(a,s,e){"use strict";e.r(s);var n=e(2),t=Object(n.a)({},(function(){var a=this,s=a.$createElement,e=a._self._c||s;return e("ContentSlotsDistributor",{attrs:{"slot-key":a.$parent.slotKey}},[e("Boxx"),a._v(" "),e("p"),e("div",{staticClass:"table-of-contents"},[e("ul",[e("li",[e("a",{attrs:{href:"#软件安装及测试"}},[a._v("软件安装及测试")]),e("ul",[e("li",[e("a",{attrs:{href:"#安装方法"}},[a._v("安装方法")])]),e("li",[e("a",{attrs:{href:"#小测试数据下载"}},[a._v("小测试数据下载")])])])]),e("li",[e("a",{attrs:{href:"#软件参数"}},[a._v("软件参数")])])])]),e("p"),a._v(" "),e("h1",{attrs:{id:"介绍"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#介绍"}},[a._v("#")]),a._v(" 介绍")]),a._v(" "),e("p",[a._v("BUSCO软件可以对基因组，注释结果进行质量评估，作者目前已经更新到了v5版本。BUSCO软件根据OrthoDB数据库，构建了几个大的进化分支的单拷贝基因集，该基因集在对应分类集合上理论上保守且仅有一个拷贝。将基因组/蛋白序列与该基因集进行比较，根据比对上的比例、完整性，来评价拼接结果的准确性和完整性。")]),a._v(" "),e("div",{staticClass:"custom-block tip"},[e("p",[a._v("仓库地址: https://busco.ezlab.org/busco_userguide.html "),e("br"),a._v("\ncite: Seppey M., Manni M., Zdobnov E.M. (2019) BUSCO: Assessing Genome Assembly and Annotation Completeness. In: Kollmar M. (eds) Gene Prediction. Methods in Molecular Biology, vol 1962. Humana, New York, NY. 2019 doi.org/10.1007/978-1-4939-9173-0_14. PMID:31020564")])]),a._v(" "),e("h2",{attrs:{id:"软件安装及测试"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#软件安装及测试"}},[a._v("#")]),a._v(" 软件安装及测试")]),a._v(" "),e("h3",{attrs:{id:"安装方法"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#安装方法"}},[a._v("#")]),a._v(" 安装方法")]),a._v(" "),e("p",[a._v("直接拉取 docker镜像（推荐）")]),a._v(" "),e("div",{staticClass:"language-sh line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-sh"}},[e("code",[a._v("docker pull ezlabgva/busco:v5.0.0_cv1\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br")])]),e("h3",{attrs:{id:"小测试数据下载"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#小测试数据下载"}},[a._v("#")]),a._v(" 小测试数据下载")]),a._v(" "),e("p",[a._v("下载酿酒酵母基因组")]),a._v(" "),e("div",{staticClass:"language-sh line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-sh"}},[e("code",[a._v(" https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.fna.gz\n "),e("span",{pre:!0,attrs:{class:"token function"}},[a._v("gzip")]),a._v(" -d GCF_000146045.2_R64_genomic.fna.gz\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br")])]),e("p",[a._v("通过镜像打开容器")]),a._v(" "),e("div",{staticClass:"language-sh line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-sh"}},[e("code",[a._v(" docker run -ti --rm --user "),e("span",{pre:!0,attrs:{class:"token variable"}},[e("span",{pre:!0,attrs:{class:"token variable"}},[a._v("$(")]),e("span",{pre:!0,attrs:{class:"token function"}},[a._v("id")]),a._v(" -u"),e("span",{pre:!0,attrs:{class:"token variable"}},[a._v(")")])]),a._v(" -v "),e("span",{pre:!0,attrs:{class:"token environment constant"}},[a._v("$PWD")]),a._v(":/busco_wd ezlabgva/busco:v5.0.0_cv1  "),e("span",{pre:!0,attrs:{class:"token function"}},[a._v("bash")]),a._v("\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br")])]),e("p",[a._v("由于v4以上版本引入了自动odb库选择和下载，直接使用基因组文件作为输入")]),a._v(" "),e("div",{staticClass:"language-sh line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-sh"}},[e("code",[a._v(" busco --in GCF_000146045.2_R64_genomic.fna --mode genome --out tmp -f\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br")])]),e("p",[a._v("弹出的info显示自动选择的库存在问题")]),a._v(" "),e("div",{staticClass:"custom-block tip"},[e("p",[a._v("INFO:   Downloading file 'https://busco-data.ezlab.org/v5/data/lineages/archaea_odb10.2021-02-23.tar.gz'")])]),a._v(" "),e("p",[a._v("下载的是古菌的库archaea_odb10，似乎不大对劲。指定自动发现真核生物相关库的参数--auto-lineage-euk")]),a._v(" "),e("div",{staticClass:"language-sh line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-sh"}},[e("code",[a._v(" busco --in GCF_000146045.2_R64_genomic.fna --mode genome --out tmp -f --auto-lineage-euk\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br")])]),e("p",[a._v("下载了真核生物的库，但通过 busco --list-datasets 命令可以看到其实有更接近的库 --\x3e saccharomycetes_odb10 。看起来这个自动判断不大靠谱的样子。")]),a._v(" "),e("p",[a._v("手动下载saccharomycetes_odb10数据库,解压后放到指定目录，在命令指定该目录位置，注意镜像中位置由于挂载可能存在变化")]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v(" busco -i GCF_000146045.2_R64_genomic.fna -l saccharomycetes_odb10 --download_path busco_downloads -o tmp -m genome -f --offline -c 10\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br")])]),e("div",{staticClass:"custom-block tip"},[e("p",[a._v("odb库下载: https://busco-data.ezlab.org/v5/data/lineages/")])]),a._v(" "),e("h2",{attrs:{id:"软件参数"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#软件参数"}},[a._v("#")]),a._v(" 软件参数")]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v('  -i FASTA FILE, --in FASTA FILE\n                        Input sequence file in FASTA format. Can be an assembled genome or transcriptome (DNA), or protein sequences from an annotated gene set.\n  -o OUTPUT, --out OUTPUT\n                        Give your analysis run a recognisable short name. Output folders and files will be labelled with this name. WARNING: do not provide a path\n  -m MODE, --mode MODE  Specify which BUSCO analysis mode to run.\n                        There are three valid modes:\n                        - geno or genome, for genome assemblies (DNA)\n                        - tran or transcriptome, for transcriptome assemblies (DNA)\n                        - prot or proteins, for annotated gene sets (protein)\n  -l LINEAGE, --lineage_dataset LINEAGE\n                        Specify the name of the BUSCO lineage to be used.\n  --auto-lineage        Run auto-lineage to find optimum lineage path\n  --auto-lineage-prok   Run auto-lineage just on non-eukaryote trees to find optimum lineage path\n  --auto-lineage-euk    Run auto-placement just on eukaryote tree to find optimum lineage path\n  -c N, --cpu N         Specify the number (N=integer) of threads/cores to use.\n  -f, --force           Force rewriting of existing files. Must be used when output files with the provided name already exist.\n  -r, --restart         Continue a run that had already partially completed.\n  -q, --quiet           Disable the info logs, displays only errors\n  --out_path OUTPUT_PATH\n                        Optional location for results folder, excluding results folder name. Default is current working directory.\n  --download_path DOWNLOAD_PATH\n                        Specify local filepath for storing BUSCO dataset downloads\n  --datasets_version DATASETS_VERSION\n                        Specify the version of BUSCO datasets, e.g. odb10\n  --download_base_url DOWNLOAD_BASE_URL\n                        Set the url to the remote BUSCO dataset location\n  --update-data         Download and replace with last versions all lineages datasets and files necessary to their automated selection\n  --offline             To indicate that BUSCO cannot attempt to download files\n  --metaeuk_parameters METAEUK_PARAMETERS\n                        Pass additional arguments to Metaeuk for the first run. All arguments should be contained within a single pair of quotation marks, separated by commas. E.g. "--param1=1,--param2=2"\n  --metaeuk_rerun_parameters METAEUK_RERUN_PARAMETERS\n                        Pass additional arguments to Metaeuk for the second run. All arguments should be contained within a single pair of quotation marks, separated by commas. E.g. "--param1=1,--param2=2"\n  -e N, --evalue N      E-value cutoff for BLAST searches. Allowed formats, 0.001 or 1e-03 (Default: 1e-03)\n  --limit REGION_LIMIT  How many candidate regions (contig or transcript) to consider per BUSCO (default: 3)\n  --augustus            Use augustus gene predictor for eukaryote runs\n  --augustus_parameters AUGUSTUS_PARAMETERS\n                        Pass additional arguments to Augustus. All arguments should be contained within a single pair of quotation marks, separated by commas. E.g. "--param1=1,--param2=2"\n  --augustus_species AUGUSTUS_SPECIES\n                        Specify a species for Augustus training.\n  --long                Optimization Augustus self-training mode (Default: Off); adds considerably to the run time, but can improve results for some non-model organisms\n  --config CONFIG_FILE  Provide a config file\n  -v, --version         Show this version and exit\n  -h, --help            Show this help message and exit\n  --list-datasets       Print the list of available BUSCO datasets\n\n')])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br"),e("span",{staticClass:"line-number"},[a._v("16")]),e("br"),e("span",{staticClass:"line-number"},[a._v("17")]),e("br"),e("span",{staticClass:"line-number"},[a._v("18")]),e("br"),e("span",{staticClass:"line-number"},[a._v("19")]),e("br"),e("span",{staticClass:"line-number"},[a._v("20")]),e("br"),e("span",{staticClass:"line-number"},[a._v("21")]),e("br"),e("span",{staticClass:"line-number"},[a._v("22")]),e("br"),e("span",{staticClass:"line-number"},[a._v("23")]),e("br"),e("span",{staticClass:"line-number"},[a._v("24")]),e("br"),e("span",{staticClass:"line-number"},[a._v("25")]),e("br"),e("span",{staticClass:"line-number"},[a._v("26")]),e("br"),e("span",{staticClass:"line-number"},[a._v("27")]),e("br"),e("span",{staticClass:"line-number"},[a._v("28")]),e("br"),e("span",{staticClass:"line-number"},[a._v("29")]),e("br"),e("span",{staticClass:"line-number"},[a._v("30")]),e("br"),e("span",{staticClass:"line-number"},[a._v("31")]),e("br"),e("span",{staticClass:"line-number"},[a._v("32")]),e("br"),e("span",{staticClass:"line-number"},[a._v("33")]),e("br"),e("span",{staticClass:"line-number"},[a._v("34")]),e("br"),e("span",{staticClass:"line-number"},[a._v("35")]),e("br"),e("span",{staticClass:"line-number"},[a._v("36")]),e("br"),e("span",{staticClass:"line-number"},[a._v("37")]),e("br"),e("span",{staticClass:"line-number"},[a._v("38")]),e("br"),e("span",{staticClass:"line-number"},[a._v("39")]),e("br"),e("span",{staticClass:"line-number"},[a._v("40")]),e("br"),e("span",{staticClass:"line-number"},[a._v("41")]),e("br"),e("span",{staticClass:"line-number"},[a._v("42")]),e("br"),e("span",{staticClass:"line-number"},[a._v("43")]),e("br"),e("span",{staticClass:"line-number"},[a._v("44")]),e("br"),e("span",{staticClass:"line-number"},[a._v("45")]),e("br")])])],1)}),[],!1,null,null,null);s.default=t.exports}}]);